### PR TITLE
Feat/testing validators maj

### PR DESCRIPTION
**Pull request by majzabukovnik**

### CODE REVIEW

File: News_Blockchain/Database.cs [hunk 1]
```diff
@@ -135,11 +143,16 @@ public Block GetRecord(string key)
...
+    /// Function returns a list of blocks from including specified index and up to latest block
+    /// </summary>
+    /// <param name="index"></param>
+    /// <returns></returns>
     public List<Block> GetLastSpecifiedBlocks(int index)
     {
         List<Block> list = new List<Block>();
         var iterator = _zoneTree.CreateIterator();
-        for (int i = 0; i < index; i++)
+        for (int i = index; i <= lastKnownBlockIndex; i++)
```
Comment (majzabukovnik) [inline News_Blockchain/Database.cs:155]: Function functionality has been changed. @bajla Confirm if it was intended to work in this way

File: News_Blockchain/BlockValidator.cs [hunk 2]
```diff
@@ -280,8 +292,7 @@ public static double CalculateBlockFees(Block block)
                     UTXODB utxoDB = new UTXODB();
```
Comment (majzabukovnik) [inline News_Blockchain/BlockValidator.cs]: @bajla This probably also needs to be injection

File: News_Blockchain/BlockValidator.cs [hunk 3]
```diff
@@ -492,5 +505,18 @@ public static Block CreateBlock(List<Transaction> transactions)
...
+
+        /// <summary>
+        /// Function creates valid transaction
+        /// </summary>
+        /// <param name="address">Recipient's address</param>
+        /// <param name="value">Value of transfer</param>
+        /// <param name="fee">Fee in satoshy/byte</param>
+        /// <param name="text">Optional argument for news content</param>
+        /// <returns></returns>
+        public static Transaction CrateTransaction(string address, double value, int fee, string text = "")
```
Comment (majzabukovnik) [inline News_Blockchain/BlockValidator.cs]: @Luka1321 Try writing this function on your own. All of the needed functionality has already been written, you just need to put Legos together

File: News_Blockchain/BlockValidator.cs (hunk 2 above)
Comment (bajla) [reply News_Blockchain/BlockValidator.cs]: yes

File: News_Blockchain/Database.cs (hunk 1 above)
Comment (bajla) [reply News_Blockchain/Database.cs:155]: what functionality?

**Comment by majzabukovnik** [comment]
Validators should work now. They haven't been tested on normal trx since I only have coinbase trx. 

TODO @bajla:
- [ ] approve this pull request
- [ ] delete console.writelines from miner 
- [ ] add code into miner for saving and sending valid blocks
- [ ] check above comments
- [ ] we need to save utxo that are ours into separate array of strings 
- [ ] frontend

File: News_Blockchain/BlockValidator.cs (hunk 2 above)
Comment (majzabukovnik) [reply News_Blockchain/BlockValidator.cs]: Injection has  been made

File: News_Blockchain/BlockValidator.cs [hunk 4]
```diff
@@ -492,5 +524,63 @@ public static Block CreateBlock(List<Transaction> transactions)
...
+                        break;
+                    }
+                }
+
+                //bug: fee value is not taken into account
+                if (sumInputs >= value)
+                    break;
+            }
+
+            double change = sumInputs - value - fee;
```
Comment (majzabukovnik) [inline News_Blockchain/BlockValidator.cs:564]: fee structure is not correct since it doesn't take into account trx size

**Review by bajla — APPROVED** [review]

**Pull request merged**